processing/src/main/java/org/apache/carbondata/processing/store/TablePage.java

```diff
@@ -99,9 +99,12 @@
     noDictDimensionPages = new ColumnPage[model.getNoDictionaryCount()];
     int tmpNumDictDimIdx = 0;
     int tmpNumNoDictDimIdx = 0;
-    for (int i = 0; i < dictDimensionPages.length + noDictDimensionPages.length; i++) {
+    for (int i = 0; i < tableSpec.getNumDimensions(); i++) {
       TableSpec.DimensionSpec spec = tableSpec.getDimensionSpec(i);
-      ColumnType columnType = tableSpec.getDimensionSpec(i).getColumnType();
+      if (spec.getSchemaDataType().isComplexType()) {
+        // partition columns are placed at the end. so, might present after complex columns
```

Comment:
Do you mean to skip all complex column and go to the last dimension?

Reply:
yes, initially also it was skipping. I will make it more easy to understand